src/audio/eq_iir.c

```diff
@@ -325,90 +342,111 @@ static int eq_iir_params(struct comp_dev *dev)
 	return 0;
 }
 
-static int iir_cmd_set_value(struct comp_dev *dev, struct sof_ipc_ctrl_data *cdata)
+static int iir_cmd_get_data(struct comp_dev *dev,
+			    struct sof_ipc_ctrl_data *cdata)
 {
 	struct comp_data *cd = comp_get_drvdata(dev);
-	int j;
-	uint32_t ch;
-	bool val;
-
-	if (cdata->cmd == SOF_CTRL_CMD_SWITCH) {
-		trace_eq_iir("mst");
-		for (j = 0; j < cdata->num_elems; j++) {
-			ch = cdata->chanv[j].channel;
-			val = cdata->chanv[j].value;
-			tracev_value(ch);
-			tracev_value(val);
-			if (ch >= PLATFORM_MAX_CHANNELS) {
-				trace_eq_iir_error("che");
-				return -EINVAL;
-			}
-			if (val)
-				iir_unmute_df2t(&cd->iir[ch]);
-			else
-				iir_mute_df2t(&cd->iir[ch]);
-		}
-	} else {
-		trace_eq_iir_error("ste");
-		return -EINVAL;
-	}
 
-	return 0;
+	size_t bs;
+	int ret = 0;
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		trace_eq("gbi");
+
+		/* Copy back to user space */
+		bs = cd->config->size;
+		if (bs > SOF_EQ_IIR_MAX_SIZE || bs < 1)
+			return -EINVAL;
+		if (!cd->config)
+			memcpy(cdata->data->data, cd->config, bs);
```

Comment:
what happens if cd->config is zero or NULL?

Reply:
It's best to to trace and return an error, good catch!